Formula/grafana.rb

```diff
@@ -0,0 +1,149 @@
+require "language/node"
+
+class Grafana < Formula
+  desc "Gorgeous metric visualizations and dashboards for timeseries databases."
+  homepage "http://grafana.org"
+  url "https://github.com/grafana/grafana/archive/v3.0.4.tar.gz"
+  sha256 "f26a374326e64a8f83c57fdf916ba1c8524dd55002dfe628b4854d05fed3715a"
+
+  head "https://github.com/grafana/grafana.git"
+
+  depends_on "go" => :build
+  depends_on "node" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    grafana_path = buildpath/"src/github.com/grafana/grafana"
+    grafana_path.install Dir["*"]
+    grafana_path.install ".jscs.json", ".jsfmtrc", ".jshintrc", ".bowerrc"
+
+    cd grafana_path do
+      inreplace "package.json", '"sass-lint": "^1.6.0"', '"sass-lint": "1.7.0"'
+      system "go", "run", "build.go", "setup"
+      system "go", "run", "build.go", "build"
+      system "npm", "install", *Language::Node.local_npm_install_args
+      system "npm", "install", "grunt-cli", *Language::Node.local_npm_install_args
+      system "node_modules/grunt-cli/bin/grunt", "build"
+    end
+
+    cp(grafana_path/"conf/sample.ini", grafana_path/"conf/grafana.ini")
+
+    bin.install grafana_path/"bin/grafana-cli"
+    bin.install grafana_path/"bin/grafana-server"
+    (grafana_path/"grafana").write(env_script)
+    chmod 0755, grafana_path/"grafana"
+    bin.install grafana_path/"grafana"
+    (etc/"grafana").mkpath
+    etc.install grafana_path/"conf/grafana.ini" => "grafana/grafana.ini"
+    pkgshare.install Dir[grafana_path/"conf", grafana_path/"public_gen", grafana_path/"vendor"]
+    mv pkgshare/"public_gen", pkgshare/"public"
```

Comment:
Might be a bit neater to just drop `public_gen` from the above line & then do:
```ruby
pkgshare.install "public_gen" => "public"
```
here instead.